src/flb_engine.c

```diff
@@ -690,8 +695,8 @@ int flb_engine_start(struct flb_config *config)
     }
 
     while (1) {
-        mk_event_wait(evl);
-        mk_event_foreach(event, evl) {
+        mk_event_wait(evl); /* potentially conditional mk_event_wait or mk_event_wait_2 based on bucket queue capacity for one shot events */
```

Comment:
Not needed. mk_event_wait is essentially idempotent. If first call to wait completes, additional subsequent calls to wait will not change the evl contents, and will not block. wait_2 is not needed here.